saleor/graphql/order/types.py

```diff
@@ -692,29 +692,33 @@ def return_voucher_translated_name(discounts) -> Optional[Money]:
 
     @staticmethod
     def resolve_billing_address(root: models.Order, info):
-        if not root.shipping_address_id:
+        def _resolve_billing_address(data):
+            user, address = data
+            requester = get_user_or_app_from_context(info.context)
+            if requestor_has_access(requester, user, OrderPermissions.MANAGE_ORDERS):
+                return address
+            return obfuscate_address(address)
+
+        if not root.billing_address_id:
             return
-        requester = get_user_or_app_from_context(info.context)
-        if requestor_has_access(requester, root.user, OrderPermissions.MANAGE_ORDERS):
-            return AddressByIdLoader(info.context).load(root.billing_address_id)
-        return (
-            AddressByIdLoader(info.context)
-            .load(root.billing_address_id)
-            .then(obfuscate_address)
-        )
+        user = UserByUserIdLoader(info.context).load(root.user_id)
```

Comment:
This dataloader will fail if there is no user assigned to this dataloader.

Reply:
We should add a similar test to `test_query_order_without_addresess`, but we should fetch the address for the order without the assigned user.

Reply:
yep, i've seen that, added in_progress label to fix it and tweak the performance further

Reply:
To make it clear: it's correct for the dataloader to fail, we should not call the loader if there is nothing to load (I don't want dataloaders to discard incorrect keys silently).